### PR TITLE
switch from Travis to GH actions

### PR DESCRIPTION
**Pull request by pawelmhm**

### CODE REVIEW

**Comment by pawelmhm** [comment]
@Gallaecio @ejulio switched from Travis to GH actions, tests passing for 3.6 until 3.9. Having working CI is an essential condition for doing any other work, e.g. on porting to 3.10. Cassettes are working fine in 3.9 and 3.10. Need to investigate why tests are failing with cassettes via vcrpy https://github.com/kevin1024/vcrpy there seems to be something with this library. Latest release of vcrpy was in October 2020, I'm not sure how active project is, perhaps we should switch to different library providing similar functionality.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        python-version: ${{matrix.python-version}}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install -r requirements-test.txt
+    - name: Run tests
+      run: |
+        pytest
```
Comment (elacuesta) [inline .github/workflows/main.yml:30]:
```suggestion
        pytest

```
Just a newline at the end of the file, for consistency

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Run tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
```
Comment (elacuesta) [inline .github/workflows/main.yml:10]:
```suggestion
  tests:
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: 
+        python-version: ["3.6", "3.7", "3.8", "3.9"]
```
Comment (elacuesta) [inline .github/workflows/main.yml:15]: Should `2.7` and `pypy` be added here?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version}}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{matrix.python-version}}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
```
Comment (elacuesta) [inline .github/workflows/main.yml:26]: This ends up installing `msgpack` for all tests, I think we should split this and test both with and without it.

File: .github/workflows/main.yml (hunk 1 above)
Comment (elacuesta) [reply .github/workflows/main.yml:30]: Also, could you push the coverage diff to codecov after running the tests?

**Review by elacuesta — COMMENTED** [review]
Looking good @pawelmhm, thanks.
Can we keep `tox.ini` just like we have for other projects? (scrapy, parsel, itemadapter come to mind). That way we can handle the msgpack tests (and other checks like flake8/black/pylint) more easily IMHO.
Lastly, it would be good to have a workflow to publish the package to PyPI. I get that the Travis build is not currently working anyway, so I don't think that should be a blocker for this PR.

**Comment by pawelmhm** [comment]
thanks for review @elacuesta I'll include your feedback in next commit

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        python-version: ${{ matrix.python-version }}
+
+    - name: Run tests
+      env: ${{ matrix.env }}
+      run: |
+        pip install -U tox
+        tox
+
+    - name: Upload coverage report
+      run: bash <(curl -s https://codecov.io/bash)
```
Comment (elacuesta) [inline .github/workflows/main.yml:68]: The bash uploader is [deprecated](https://docs.codecov.com/docs/about-the-codecov-bash-uploader), could you use the [newer one](https://docs.codecov.com/docs/codecov-uploader)?

File: .github/workflows/publish.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+    - name: Build distribution
+      run: |
+        pip install -U setuptools wheel
+        python setup.py sdist bdist_wheel
+    
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.PYPI_TOKEN }}
```
Comment (elacuesta) [inline .github/workflows/publish.yml:26]: @Gallaecio would you be able to get a token for this repo?

**Comment by pawelmhm** [comment]
@elacuesta 

- added pypy and python 2.7
- codecov uplodad[ via github action](https://github.com/marketplace/actions/codecov)
- publish yml action (launched on new release in github, should be tested properly)
- brought back tox

**Comment by pawelmhm** [comment]
About PYPI_TOKEN I don't have permissions to create one so we need to wait for @Gallaecio, would be also nice to have token to testpypi to test it first on first release via github.

**Review by elacuesta — APPROVED** [review]
Awesome, many thanks @pawelmhm. The PyPI token was added to the repo secrets some time ago by @Gallaecio, we don't need to worry about it.

**Review by wRAR — APPROVED** [review]

**Pull request merged**